server/hotrod/src/main/java/org/infinispan/server/hotrod/streaming/StreamingState.java

```diff
@@ -0,0 +1,64 @@
+package org.infinispan.server.hotrod.streaming;
+
+import org.infinispan.metadata.Metadata;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.Channel;
+
+public abstract class StreamingState {
```

Comment:
I can refactor this to have no parent class and have two maps instead. This originally had more overlap in methods and has diverged as I fixed everything. Just let me know what you think.